lib/iris/fileformats/pp.py

```diff
@@ -1513,9 +1545,34 @@ def t2(self, dt):
         self.lbhrd = dt.hour
         self.lbmind = dt.minute
         self.lbdayd = int(dt.strftime("%j"))
+        if self.lbyrd == 0:
+            if self.lbmond == 0 and self.lbdatd == 0:
+                self.lbmond = self.lbdatd = 1
+            elif self.lbmond == 0 or self.lbdatd == 0:
+                emsg = (
+                    f"Invalid {self.__class__.__name__} t2 date {dt} provided"
+                )
+                raise ValueError(emsg)
         if hasattr(self, "_t2"):
             delattr(self, "_t2")
 
+    @property
+    def t2_zero(self):
+        """
+        Initialise the associated ``t2`` field properties to zero.
+
+        It is not possible to do this though the setter property as true zero
+        dates are not valid :func:`datetime.datetime` or :func`cftime.datetime`
+        dates.
+
+        """
+        self.lbyrd = 0
+        self.lbmond = 0
+        self.lbdatd = 0
+        self.lbhrd = 0
+        self.lbmind = 0
+        self.lbdayd = 0
+
```

Comment:
I don't know if it helps here, but calendar agnostic datetimes are still possible at `cftime` v1.5 e.g.

```python
cftime.datetime(0, 0, 0, calendar=None)
```

Reply:
Don't know either, but I'd imaging holding on to the `calendar` is more important rather than being able to represent an invalid date.

Most of the fixes throughout this PR involve baking the `calendar` back into the `cftime.datetime`, because now it's possible.

Reply:
> I can perhaps give it a try and see what happens...

Maybe wait for a proper reviewer who understands this module!

Reply:
@rcomer To be honest, that was a lovely nugget of a simplification that I've now applied, thanks :star_struck: